contrib/node/src/python/pants/contrib/node/tasks/javascript_style.py

```diff
@@ -0,0 +1,122 @@
+# coding=utf-8
+# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnitLabel
+from pants.util.contextutil import pushd
+from pants.util.memo import memoized_method
+
+from pants.contrib.node.targets.node_package import NodePackage
+from pants.contrib.node.tasks.node_task import NodeTask
+
+
+class JavascriptStyle(NodeTask):
+  """ Check javascript source files to ensure they follow the style guidelines.
+
+  :API: public
+  """
+
+  _JS_SOURCE_EXTENSION = '.js'
+  _JSX_SOURCE_EXTENSION = '.jsx'
+  INSTALL_JAVASCRIPTSTYLE_TARGET_NAME = 'synthetic-install-javascriptstyle-module'
+
+  def __init__(self, *args, **kwargs):
+    super(JavascriptStyle, self).__init__(*args, **kwargs)
+
+  @classmethod
+  def register_options(cls, register):
+    super(JavascriptStyle, cls).register_options(register)
+    register('--skip', type=bool, fingerprint=True, help='Skip javascriptstyle.')
+    register('--fail-slow', type=bool,
+             help='Check all targets and present the full list of errors.')
+    register('--javascriptstyle-dir', advanced=True, fingerprint=True,
+             help='Package directory for lint tool.')
+
+  def get_lintable_node_targets(self, targets):
+    return filter(
+      lambda target: isinstance(target, NodePackage)
+                     and (target.has_sources(self._JS_SOURCE_EXTENSION)
+                          or target.has_sources(self._JSX_SOURCE_EXTENSION))
+                     and (not target.is_synthetic),
+      targets)
+
+  def get_javascript_sources(self, target):
+    sources = set()
+    sources.update(os.path.join(get_buildroot(), source) for source in target.sources_relative_to_buildroot()
+                   if (source.endswith(self._JS_SOURCE_EXTENSION) or
+                       source.endswith(self._JSX_SOURCE_EXTENSION)))
+    return sources
+
+  @memoized_method
+  def _install_javascriptstyle(self):
+    javascriptstyle_dir = self.get_options().javascriptstyle_dir
+    with pushd(javascriptstyle_dir):
```

Comment:
When we package up pants, that directory will not come along with it, so this task will explode by default (until someone configures that option).

If you were to handle `self.get_options().javascriptstyle_dir is None`, and warn that the task is being skipped because it is not configured, then that might be ok. But it isn't particularly clear what someone should install in that directory, and I think that currently this will lead to a lot of copy-pasta.

My suggestion would be to try documenting what a user would do to set up lint in their repo: if the instructions end up long and confusing (as I think they might here) then you might consider taking a different tack, such as having an option that references a Yarn package version, or having the tool default to using a `node_module` target tool (which could then be configured in either way). The advantage of both of these options relative to what you're doing here is that they can have reasonable defaults.